pkg/sbom/v1/sbom_spdx_storage_format.go

```diff
@@ -22,39 +26,80 @@ const (
 )
 
+var spdxDataDirPath string
+
 type SBOMData struct {
-	spdxData                              spdxv1beta1.SBOMSPDXv2p3
+	spdxDataPath                          string
 	filteredSpdxData                      spdxv1beta1.SBOMSPDXv2p3Filtered
 	relevantRealtimeFilesBySPDXIdentifier sync.Map
 	newRelevantData                       bool
 	alreadyExistSBOM                      bool
+	instanceID                            instanceidhandler.IInstanceID
+}
+
+func init() {
+	wd, err := os.Getwd()
+	if err != nil {
+		logger.L().Ctx(context.GetBackgroundContext()).Error("failed to get working directory", helpers.Error(err))
+	}
+	spdxDataDirPath = fmt.Sprintf("%s/%s", wd, directorySBOM)
+	err = os.MkdirAll(spdxDataDirPath, os.ModeDir|os.ModePerm)
+	if err != nil {
+		logger.L().Ctx(context.GetBackgroundContext()).Error("failed to create directory for SBOM resources", helpers.String("directory path", spdxDataDirPath), helpers.Error(err))
+	}
 }
 
-func CreateSBOMDataSPDXVersionV040() *SBOMData {
+func CreateSBOMDataSPDXVersionV040(instanceID instanceidhandler.IInstanceID) SBOMFormat {
+
 	return &SBOMData{
+		spdxDataPath:                          fmt.Sprintf("%s/%s", spdxDataDirPath, instanceID.GetHashed()),
 		filteredSpdxData:                      spdxv1beta1.SBOMSPDXv2p3Filtered{},
 		relevantRealtimeFilesBySPDXIdentifier: sync.Map{},
 		newRelevantData:                       false,
 		alreadyExistSBOM:                      false,
+		instanceID:                            instanceID,
 	}
 }
 
+func (sbom *SBOMData) saveSBOM(spdxData *spdxv1beta1.SBOMSPDXv2p3) error {
+	f, err := os.Create(sbom.spdxDataPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	data, err := json.Marshal(spdxData)
+	if err != nil {
+		return err
+	}
+	_, err = f.Write(data)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 func (sbom *SBOMData) StoreSBOM(sbomData any) error {
 	spdxData, ok := sbomData.(*spdxv1beta1.SBOMSPDXv2p3)
 	if !ok {
 		return fmt.Errorf("storage format: StoreSBOM: SBOM data format is not supported")
 	}
 
-	sbom.spdxData = *spdxData
-	for i := range sbom.spdxData.Spec.SPDX.Files {
-		sbom.relevantRealtimeFilesBySPDXIdentifier.Store(spdxv1beta1.ElementID(sbom.spdxData.Spec.SPDX.Files[i].FileSPDXIdentifier), false)
+	err := sbom.saveSBOM(spdxData)
+	if err != nil {
+		return err
+	}
+
+	for i := range spdxData.Spec.SPDX.Files {
+		sbom.relevantRealtimeFilesBySPDXIdentifier.Store(spdxv1beta1.ElementID(spdxData.Spec.SPDX.Files[i].FileSPDXIdentifier), false)
 	}
 
-	sbom.filteredSpdxData.Spec = sbom.spdxData.Spec
-	sbom.filteredSpdxData.Status = sbom.spdxData.Status
-	sbom.spdxData.Spec.SPDX.CreationInfo.Creators = append(sbom.spdxData.Spec.SPDX.CreationInfo.Creators, []spdxv1beta1.Creator{
+	sbom.filteredSpdxData.Spec = spdxData.Spec
+	sbom.filteredSpdxData.Status = spdxData.Status
+	sbom.filteredSpdxData.Spec.SPDX.CreationInfo.Creators = append(sbom.filteredSpdxData.Spec.SPDX.CreationInfo.Creators, []spdxv1beta1.Creator{
```

Comment:
Can you please explain what was changed here?
Are we accessing a different object?
And why are there no interfaces here? what if we change the SPDX format?

Reply:
this all file is an implementation for our SPDX format - the interface is in the same dir